test/studies/shootout/fannkuch-redux/bradc/fannkuch-redux-blc.chpl

```diff
@@ -1,78 +1,119 @@
 /* The Computer Language Benchmarks Game
- * http://benchmarksgame.alioth.debian.org/
- *
- * contributed by Kyle Brady and Brad Chamberlain
- * based on C implementation by Ledrug Katz
- *
- */
+   http://benchmarksgame.alioth.debian.org/
 
-config const N = 7;
+   contributed by Ben Harshbarger and Brad Chamberlain
+   derived from the Swift implementation by Ralph Ganszky
+*/
 
-assert(N >=3, "N must be in 3..");
+use DynamicIters;
+
+config const n = 7,          // the array size over which to compute perms
+             nchunks = 720;  // the number of chunks of parallelism
+
+//
+// memoize n! (n factorial)
+//
+var fact: [0..n] int;
+fact[0] = 1;
+for i in 1..n do
+  fact[i] = i*fact[i-1];
```

Comment:
Why do we want to initialize fact in the module initializer as opposed to main?  This strikes me as potentially confusing to new users trying to understand the order of execution for bits of the file (even with this initialization occurring before the main function is encountered).